docs/website/docs/developers/design-docs/hip-hal-driver.md

```diff
@@ -0,0 +1,263 @@
+---
+hide:
+  - tags
+tags:
+  - GPU
+  - HIP
+---
+
+# HIP HAL driver
+
+This document lists technical details regarding the HIP implemenation of
+IREE's Hardware Abstraction Layer, called a HIP HAL driver.
+
+IREE provides a [Hardware Abstraction Layer (HAL)][iree-hal] as a common
+interface to different compute accelerators. IREE HAL's design draws inspiration
+from modern GPU architecture and APIs; so implementing a HAL driver using HIP
+is mostly straightforward; though there are places we need emulation given no
+direct mapping concepts or mechanisms. HIP HAL driver draws inspiration from the
+CUDA HAL driver and the code structure is based off of that implementation.
+
+## Overall design choices
+
+### HIP driver vs runtime API
+
+IREE HAL's design draws inspiration from modern GPU APIs--it provides explicit
+control of low-level GPU objects. The compiler is expected to plan the object
+lifetime and schedule workload and synchronization in an optimized way; IREE
+HAL implementation and the underlying GPU driver stack is expected to be a thin
+layer without much smarts and magic.
+
+Unlike CUDA, HIP doesn't provide two separate API's with the same functionality
+in the name of driver and runtime. Instead it extends the HIP API with Modules
+and Ctx control API's that the CUDA driver API's exclusively offer.At runtime
+the HIP HAL driver will load the `libamdhip64.so`/`amdhip64.dll` library dynamically.
+
+## GPU Objects
+
+### Driver
```

Comment:
For a next rev, I'm working on landing: https://github.com/openxla/iree/pull/16766

Then this section will be more explicit that the Driver corresponds to the loaded libamdhip64.so. And it can be configured with either a --hip_dylib_path flag or `IREE_HIP_DYLIB_PATH` (or soon corresponding Python API).